### DISCUSSION TITLE
Uno platform mobile app does not work on iPhones

### DISCUSSION THREAD

**Discussion by Sankar-san** [unoplatform/uno#12212] [General]
We created the Uno platform project using the visual studio tool. Uno mobile app runs on web browsers, simulators and android devices but it does not work on iphone devices.

**Error details**

- [ ] ApplicationVerificationFailed: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.nErsMe/extracted/sample.Mobile.app : 0xe800801c (No code signature found.)

- [ ] error MT1006: Could not install the application '/Users/developer/Workspace/sankar/unoPlatform/sample/sample.Mobile/bin/Release/net7.0-ios/iossimulator-x64/sample.Mobile.app' on the device 'iPhone (2)': AMDeviceSecureInstallApplicationBundle returned: 0xe800801c.

- [ ] AMDeviceSecureInstallApplicationBundle returned: 0xe800801c

- [ ] Application could not be uploaded to the device.

![Screenshot 2023-05-09 at 10 38 07 AM](https://user-images.githubusercontent.com/132877139/237024320-66f45da7-bb39-4ec4-a6cc-9d59f4103209.png)

![Screenshot 2023-05-09 at 10 43 17 AM](https://user-images.githubusercontent.com/132877139/237024361-49da26d6-c5eb-4d9b-9bd1-c6c2fcc59851.png)

Could you please help me figure out how to resolve this build issue?

**Comment by spouliot** [comment]
This is strange. At first it looks like some issue with code signing.

But your 2nd screenshot shows a `Release` build for devices, while your 1st screen shot mention the app being built for `iossimulator-x64`. That of course cannot work.

Can you attach a `.binlog` of the build ?
Or attach a zipped version of your project ?

**Reply by Sankar-san** [reply]
![Screenshot 2023-05-10 at 1 34 03 PM](https://github.com/unoplatform/uno/assets/132877139/72b2d303-9942-4021-97df-5146630274ba)


[build_log.txt](https://github.com/unoplatform/uno/files/11439644/build_log.txt)

Build log file screens.
I changed the build options Release to Release X64.

**Reply by spouliot** [reply]
The attached build is not a `.binlog` so it does not have as much information. It's also the log of a successful build - so it does not show anything wrong.

A `x64` release build seems weird - since there's no iPhone with an intel x64 processor. However your screenshot still mention `simulator-x64` while the UI hints you are targeting a physical device (and not a simulator).

**Reply by Sankar-san** [reply]
```
Target ILLink:
    /usr/local/share/dotnet/sdk/7.0.203/Sdks/Microsoft.NET.ILLink.Tasks/build/Microsoft.NET.ILLink.targets(86,5): error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false.
Done building target "ILLink" in project "sample.Mobile.csproj" -- FAILED.

Done building project "sample.Mobile.csproj" -- FAILED.

Build FAILED.

/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.7054/targets/Xamarin.Shared.Sdk.targets(1494,3): warning : The file '/Users/developer/.nuget/packages/uno.fonts.fluent/2.0.0/lib/netstandard1.0/Uno.Fonts.Fluent.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle.
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/16.4.7054/targets/Xamarin.Shared.Sdk.targets(1494,3): warning : The file '/Users/developer/.nuget/packages/uno.fonts.roboto/2.1.0-dev.9/lib/netstandard2.0/Uno.Fonts.Roboto.uprimarker' does not specify a 'PublishFolderType' metadata, and a default value could not be calculated. The file will not be copied to the app bundle.
ILLink : unknown error IL7000: An error occured while executing the custom linker steps. Please review the build log for more information.
ILLINK : error MT2301: The linker step 'Setup' failed during processing: An item with the same key has already been added. Key: MONO_GC_PARAMS
/usr/local/share/dotnet/sdk/7.0.203/Sdks/Microsoft.NET.ILLink.Tasks/build/Microsoft.NET.ILLink.targets(86,5): error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false.
    2 Warning(s)
    3 Error(s)

Time Elapsed 00:00:07.45
```

Above error showing from build log window. 

Could you please tell me about references for ios app step by step configuration guidance?

**Reply by spouliot** [reply]
This does not look like the previous issues you mentioned. You need to provide either a self contained test case (so we can reproduce the issue) and/or the binary logs so we can investigate any problem.
